src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -876,21 +876,29 @@ bool ReadMemoryValueFromFile(const char* filename, uint64_t* val)
     return result;
 }
 
+#define CHECK_CACHE_SIZE(CACHE_LEVEL) if (size > cacheSize) { cacheSize = size; cacheLevel = CACHE_LEVEL; }
+
 static size_t GetLogicalProcessorCacheSizeFromOS()
 {
+    size_t cacheLevel = 0;
     size_t cacheSize = 0;
+    size_t size;
 
 #ifdef _SC_LEVEL1_DCACHE_SIZE
-    cacheSize = std::max(cacheSize, ( size_t) sysconf(_SC_LEVEL1_DCACHE_SIZE));
+    size = ( size_t) sysconf(_SC_LEVEL1_DCACHE_SIZE);
```

Comment:
Q (also applies to the old src) - what does sysconf return if there's no cache of that level? I presume it returns 0, not -1 (the doc says it returns -1 if there's an error).